earth_enterprise/src/common/geFilePool.cpp

```diff
@@ -20,17 +20,65 @@
 #include <third_party/rsa_md5/crc32.h>
 #include <khEndian.h>
 
+// ****************************************************************************
+// ***  POSIXFileAcessor
+// ****************************************************************************
+int POSIXFileAccessor::Open(const std::string &fname, int flags, mode_t createMask) {
+  return khOpen(fname, flags, createMask);
+}
+
+int POSIXFileAccessor::FsyncAndClose(AbstractFileIdentifier* pID) {
+  return khFsyncAndClose(pID->getAsFD());
+}
+
+int POSIXFileAccessor::Close(AbstractFileIdentifier* pID) {
+  return khClose(pID->getAsFD());
+}
+
+bool POSIXFileAccessor::PreadAll(AbstractFileIdentifier* pID, void* buffer, size_t size, off64_t offset) {
+  return khPreadAll(pID->getAsFD(), buffer, size, offset);
+}
+
+bool POSIXFileAccessor::PwriteAll(AbstractFileIdentifier* pID, const void* buffer, size_t size, off64_t offset) {
+  return khPwriteAll(pID->getAsFD(), buffer, size, offset);
+}
+
+// ****************************************************************************
+// ***  FileIdentifierFactory
+// ****************************************************************************
+namespace FileIdentifierFactory {
+  static AbstractFileIdentifier* getIdentifier(int i) {
+    return new POSIXIdentifier{i};
+  }
+}
+
+// ****************************************************************************
+// ***  FileAccessorFactory
+// ****************************************************************************
+namespace FileAccessorFactory {
+  static AbstractFileAccessor* getAccessor(const std::string &fname) {
```

Comment:
Similar to the suggestion above, it would be better to have the return type be `shared_ptr<AbstractFileAccessor>` nd return `std::make_shared<POSIXFileAccessor>()`.

Same for the `getAccessor` function below.